loader/loader.c

```diff
@@ -897,7 +897,7 @@ bool loader_implicit_layer_is_enabled(const struct loader_instance *inst, const
     // The disable_environment has priority over everything else.  If it is defined, the layer is always
     // disabled.
     env_value = loader_getenv(prop->disable_env_var.name, inst);
-    if (env_value) {
```

Comment:
No, i meant to do it here.  888 is the "enable" environment variable and I'm not entirely sure what the heck `prop->enable_env_var.value` is supposed to be doing there.

This is the "disable".  If it's defined but 0, it's the same as not being defined.  So either it has to be defined, or defined to a non-0 value for the layer to be disabled.

Reply:
Except, that differs from the docs, which say:

> Implicit Layers Only - REQUIRED: Indicates an environment variable used to disable the Implicit Layer (when defined to any non-empty string value).

`prop->enable_env_var.value` is needed because the layer manifest is what defines which value should enable the env-var.

> This environment variable (which may vary with each variation of the layer) must be set to the given value or else the implicit layer is not loaded. 

Thus, we load the manifest, check if there is an `enable_environment` then try to load that env-var. If it exists, and the value doesn't match what is in the manifest, don't enable the layer.

Reply:
Someone changed the disable docs.  It used to say the value was defined to non-zero several years ago (or should have).  I'll remove the atoi check then.

Reply:
But, this also contradicts every other usage of the environment variables in the loader, just sayin.

Reply:
It seems like it has been this way for a looong time.
https://github.com/KhronosGroup/Vulkan-Loader/commit/274e4bcf33caeb0f22f59efe6bf2adf44481bba2